Code/Mantid/Framework/WorkflowAlgorithms/src/ConvolutionFitSequential.cpp

```diff
@@ -0,0 +1,634 @@
+ * @param subFunction The unchecked substring of the function
+ * @return true if the function is two lorentzian false if one lorentzian
+ */
+bool ConvolutionFitSequential::checkForTwoLorentz(
+    const std::string &subFunction) {
+  std::string fitType = "";
+  auto pos = subFunction.rfind("name=");
+  if (pos != std::string::npos) {
+    fitType = subFunction.substr(pos, subFunction.size());
+    pos = fitType.find_first_of(",");
+    fitType = fitType.substr(5, pos - 5);
+    if (fitType.compare("Lorentzian") == 0) {
+      return true;
+    }
+  }
+  return false;
+}
+
+/**
+ * Finds specific values embedded in the function supplied to the algorithm
+ * @param function The full function string
+ * @return all values of interest from the function (0 - fitType,  1 -
+ * functionName)
+ */
+std::vector<std::string>
+ConvolutionFitSequential::findValuesFromFunction(const std::string &function) {
+  std::vector<std::string> result;
+  std::string fitType = "";
+  std::string functionName = "";
+  auto startPos = function.rfind("name=");
+  if (startPos != std::string::npos) {
+    fitType = function.substr(startPos, function.size());
+    auto nextPos = fitType.find_first_of(",");
+    fitType = fitType.substr(5, nextPos - 5);
+    functionName = fitType;
+    if (fitType.compare("Lorentzian") == 0) {
+      std::string newSub = function.substr(0, startPos);
+      bool isTwoL = checkForTwoLorentz(newSub);
+      if (isTwoL == true) {
+        fitType = "2";
+      } else {
+        fitType = "1";
+      }
+    } else {
+      fitType = "0";
+    }
+    result.push_back(fitType);
+  }
+
+  result.push_back(functionName);
+  return result;
+}
+
+/**
+ * Searchs for a given fit parameter within the a vector of columnNames
+ * @param suffix - The string to search for within the columnName
+ * @param columns - A vector of column names to be searched through
+ * @return A vector of all the column names that contained the given suffix
+ * string
+ */
+std::vector<std::string> ConvolutionFitSequential::searchForFitParams(
+    const std::string &suffix, const std::vector<std::string> &columns) {
+  auto fitParams = std::vector<std::string>();
+  const size_t totalColumns = columns.size();
+  for (size_t i = 0; i < totalColumns; i++) {
+    auto pos = columns.at(i).rfind(suffix);
+    if (pos != std::string::npos) {
+      auto endCheck = pos + suffix.size();
+      if (endCheck == columns.at(i).size()) {
+        fitParams.push_back(columns.at(i));
+      }
+    }
+  }
+  return fitParams;
+}
+
+/**
+ * Inline function to Square a number
+ */
+static inline double computeSquare(double x) { return x * x; }
+
+/**
+ * Squares all the values inside a vector of type double
+ * @param target - The vector to be squared
+ * @return The vector after being squared
+ */
+std::vector<double>
+ConvolutionFitSequential::squareVector(std::vector<double> target) {
+  std::transform(target.begin(), target.end(), target.begin(), computeSquare);
```

Comment:
Good use of the STL here. Only a minor point that there is a `Squares` class in `Kernel/VectorHelper.h` that you can use instead a custom static function:
```
std::transform(target.begin(), target.end(), target.begin(), Squares());
```